src/System.Net.HttpListener/tests/HttpListenerFactory.cs

```diff
@@ -7,15 +7,34 @@
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
 using System.Text;
+using System.IO;
 
 namespace System.Net.Tests
 {
     // Utilities for generating URL prefixes for HttpListener
     public class HttpListenerFactory : IDisposable
     {
-        const int MinPort = 1025;
-        private static readonly object s_createListenerLock = new object();
-        private static int s_nextPort = MinPort;
+        const int MaxStartAttempts = 50;
+        const string ProcPortRangeFileName = "/proc/sys/net/ipv4/ip_local_port_range";
+        private static int _minPort;
+        private static int _maxPort;
+        private static readonly Random _random = new Random();
+
+        static HttpListenerFactory()
+        {
+            // Chose a port from the dynamic port range. The IANA recommended range is 49152-65535.
+            _minPort = 49152;
+            _maxPort = 65535;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && File.Exists(ProcPortRangeFileName))
+            {
+                string[] limits = File.ReadAllText(ProcPortRangeFileName).Split(new[] { '\t', '\n' }, 2, StringSplitOptions.RemoveEmptyEntries);
+                if (limits.Length == 2)
+                {
+                    int.TryParse(limits[0], out _minPort);
+                    int.TryParse(limits[1], out _maxPort);
```

Comment:
If these TryParses fail, won't _minPort and/or _maxPort be set to 0?  Seems like you'd want these to be:
```C#
int tmp;
if (int.TryParse(limits[0], out tmp) _minPort = tmp;
if (int.TryParse(limits[1], out tmp) _maxPort = tmp;
```